src/App.tsx

```diff
@@ -15,36 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically,
+  Length,
+  Reset,
+}
+
+export function sortGoods(
+  goods: string[],
+  type: SortType,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  switch (type) {
+    case SortType.Alphabetically:
+      preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      break;
+    case SortType.Length:
+      preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
```

Comment:
The function name `sortGoods` should be more descriptive by starting with a verb. Consider renaming it to `sortGoodsByType` or similar to indicate what it does.